python/ray/data/_internal/dataset_iterator_impl.py

```diff
@@ -52,6 +56,35 @@ def iter_batches(
 
         stats.iter_total_s.add(time.perf_counter() - time_start)
 
+    def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]:
+        # During row-based ops, we also choose a batch format that lines up with the
+        # current dataset format in order to eliminate unnecessary copies and type
+        # conversions.
+        ctx = DatasetContext.get_current()
```

Comment:
This part is duplicating code of Dataset.iter_rows(). Should we just call it or make Dataset.iter_rows() call this one?

Reply:
This PR already changes it so that `dataset.iter_rows()` calls into this: https://github.com/ray-project/ray/pull/33180/files#diff-5a812d24fc3efd599284bb40dcbd2c6627bbe324e1c89a6e623b4dccf8e5de44R2858

Following the same patter we do for `iter_batches`